src/main/java/com/theironyard/controllers/GoodreadsController.java

```diff
@@ -0,0 +1,170 @@
+package com.theironyard.controllers;
+
+import com.theironyard.models.Book;
+import com.theironyard.models.User;
+import com.theironyard.repositories.BookRepository;
+import com.theironyard.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.servlet.http.HttpSession;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by melmo on 12/29/16.
+ */
+@Controller
+public class GoodreadsController {
+    public static final String SESSION_USER_ID = "currentUserId";
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    BookRepository bookRepository;
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model, HttpSession session, @RequestParam(defaultValue = "0") int allPage, @RequestParam(defaultValue = "0") int userPage, String filterUser, String filterBy, String sortUser, String sortBy, String filterAll, String sortAll){
+        Integer currentUserId = (Integer) session.getAttribute(SESSION_USER_ID);
+        String returnVal;
+        Page<Book> usersBooks = null;
+        Page<Book> allBooks = null;
+
+        if (currentUserId != null){
+            User currentUser = userRepository.findOne(currentUserId);
+            if (Boolean.valueOf(filterUser)){
+                usersBooks = bookRepository.findByUserAndStatus(currentUser, filterBy, new PageRequest(userPage, 5));
+            }
+            else if (Boolean.valueOf(sortUser)){
+                if (sortBy.equals("title")){
+                    usersBooks = bookRepository.findByUserOrderByTitleAsc(currentUser, new PageRequest(userPage, 5));
```

Comment:
There is a version of these methods that take a `Sort` object that would greatly reduce the code by simply creating the sort object and passing it to 1 method in the repository.